crates/ty_python_semantic/src/types.rs

```diff
@@ -669,6 +669,10 @@ impl<'db> Type<'db> {
         matches!(self, Type::Dynamic(_))
     }
 
+    pub const fn is_typed_dict(&self) -> bool {
```

Comment:
nit: I think we should avoid making things `pub` wherever possible, as Clippy doesn't warn us about unused code if something is `pub`. (I know there are lots of `Type` methods that violate this general rule, and ideally we'd clean them up IMO, but it's also probably not that important to do so.)

```suggestion
    pub(crate) const fn is_typed_dict(&self) -> bool {
```